rustworkx/__init__.py

```diff
@@ -15,7 +16,8 @@
 # flake8: noqa
 import rustworkx.visit
 
-sys.modules["rustworkx.generators"] = generators
+__rustworkx_mod__ = importlib.import_module(".rustworkx", package="rustworkx")
+sys.modules["rustworkx.generators"] = __rustworkx_mod__.generators
```

Comment:
Out of curiosity, why `importlib` over the regular `from . import rustworkx as __rustworkx_mod__`?

Reply:
It is a matter of personal preference, but in my opinion it pairs better with `sys.modules`. For `rustworkx.generators`, the old version was not readable at all. `importlib.import_module` is a common pair with `sys.modules`

Reply:
I'm not really sure I follow - `importlib.import_module` is kind of just a slightly less efficient way of doing retrieving the exact same module object when all the names are literals - but it was mostly idle curiosity and not meant to be an issue. I'd just never have thought to use the dynamic system when the static system would work.

Reply:
It is just that:
```
__rustworkx_mod__ = importlib.import_module(".rustworkx", package="rustworkx")
sys.modules["rustworkx.generators"] = __rustworkx_mod__.generators # clear where this came from
```
Is less confusing them:
```
from .rustworkx import *
sys.modules["rustworkx.generators"] = generators  # where did this variable come from???
```

Reply:
That second one isn't quite what I'm suggesting - I'm saying the second would be
```python
from . import rustworkx as __rustworkx_mod__
from .rustworkx import *
sys.modules["rustworkx.generators"] = __rustworkx_mod__.generators
```

No `importlib` needed.

Reply:
I will close as `from . import rustworkx` is equally as cryptic and we came to the same conclusion